federation-js/src/composition/compose.ts

```diff
@@ -176,6 +183,34 @@ export function buildMapsFromServiceList(serviceList: ServiceDefinition[]) {
             );
           }
         }
+
+        for (const field of definition.fields ?? []) {
+          const fieldName = field.name.value;
+          const tagDirectives = findDirectivesOnNode(field, 'tag');
+
+          // TODO: we can dry this up into one step for both cases
+          if (tagDirectives.length > 0) {
+            const fieldToDirectivesMap = mapGetOrSet(
+              typeNameToFieldDirectivesMap,
+              typeName,
+              new Map(),
+            );
+            const directives = mapGetOrSet(fieldToDirectivesMap, fieldName, []);
+            directives.push(...tagDirectives);
+          }
+          const inaccessibleDirectives = findDirectivesOnNode(field, 'inaccessible');
+          if (inaccessibleDirectives.length === 1) {
+            const fieldToDirectivesMap = mapGetOrSet(
+              typeNameToFieldDirectivesMap,
+              typeName,
+              new Map(),
+            );
+            const directives = mapGetOrSet(fieldToDirectivesMap, fieldName, []);
+            directives.push(inaccessibleDirectives[0]);
+          } else if (inaccessibleDirectives.length > 1) {
+            // TODO: warn about multiple applications of inaccessible?
```

Comment:
I don't think we need to do that here, that should have been caught by validation of the subgraph schema.

Reply:
And if it slips through for some reason, it would be caught by validation of the supergraph :)

Reply:
Yeah, agree. Didn't think anything would be needed here but left a placeholder to come back to and be sure. Thanks for confirming.
7c32cfbe